test/utils/FinancialDataReadsUtilsSpec.scala

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package utils
+
+import play.api.libs.json.{JsError, JsPath, Json, Reads}
+import support.UnitSpec
+import v1.models.response.common.TaxPeriod
+
+class FinancialDataReadsUtilsSpec extends UnitSpec with FinancialDataReadsUtils {
+
+  case class TestItem(aField: String)
+
+  object TestItem {
+    implicit val reads: Reads[TestItem] = Json.reads[TestItem]
+  }
+
+  case class TestWrapper(items: Seq[TestItem])
+
+  object TestWrapper{
+    implicit val reads: Reads[TestWrapper] =
+      (JsPath \ "array")
+        .read(filterNotArrayReads[TestItem]("aField", Seq("notavalue")))
+        .map(TestWrapper(_))
+  }
+
+  "itemsCheck" when {
```

Comment:
Is itemsCheck necessary as it seems like a convoluted way of doing .isDefined, it's also only used in one place, so I'm not aware if there's a higher purpose for it? it's not wrong though

`  def itemsCheck[A](items : Option[Seq[A]]): Boolean = items match {
    case Some(_) => true
    case None => false
  }`

Reply:
I think this is left over from when I was doing the check for empty 'Payment' items at this level instead of in the lower level reads.